### PR TITLE
Remove cached version of queryGraphQL

### PR DESCRIPTION
**Pull request by efritz**
We should move all caching to the provider level, not on individual requests. The only place we were caching previously was for LSIF results, but the pages for references can be large.

We aren't able to cache search results due to timeouts that were affecting large monorepo customers, so the previous caching strategy was already lacking.

### CODE REVIEW

File: shared/lsif/graphql.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { LocationConnectionNode, nodeToLocation } from './conversion'
  * relatively low unless it's a VERY popular library and LSIF data is
  * ubiquitous (which is our goal).
  */
-const MAX_REFERENCE_PAGE_REQUESTS = 20
+const MAX_REFERENCE_PAGE_REQUESTS = 5
```
Comment (creachadair) [inline shared/lsif/graphql.ts]: No objection, but why this change? Does removal of caching make the previous limit bad?

**Review by creachadair — APPROVED** [review]

File: shared/lsif/graphql.ts (hunk 1 above)
Comment (efritz) [reply shared/lsif/graphql.ts]: No, this is kind of a drive-by.

Comment (efritz) [reply shared/lsif/graphql.ts]: Actually I'll revert this line.
